### PR TITLE
Add constant atoms/integers and bitstrings.

### PR DESCRIPTION
**Pull request by NJichev**

### CODE REVIEW

File: test/stream_data/types_test.exs [hunk 1]
```diff
@@ -78,6 +78,74 @@ defmodule StreamData.TypesTest do
...
+    end
+
+    test "bitstrings with size 1 and unit 8" do
+      data = generate_data(:literal_size_1_unit_8)
+
+      check all x <- data do
+        assert <<_::1*8>> = x
+      end
+    end
+
```
Comment (whatyouhide) [inline test/stream_data/types_test.exs]: This line should be gone.

File: test/stream_data/types_test.exs [hunk 2]
```diff
@@ -78,6 +78,74 @@ defmodule StreamData.TypesTest do
...
+      check all x <- data do
+        assert x == ""
+      end
+    end
+
+    test "bitstrings with size 1 and unit 8" do
+      data = generate_data(:literal_size_1_unit_8)
+
+      check all x <- data do
+        assert <<_::1*8>> = x
```
Comment (whatyouhide) [inline test/stream_data/types_test.exs]: This passes only for binaries with 8 bits in them, right? But the typespec you defined means a binary with one bit plus any number of **bytes**.

File: lib/stream_data_types.ex [hunk 3]
```diff
@@ -211,7 +211,28 @@ defmodule StreamDataTypes do
...
+    units = map(non_negative_integer(), &(&1 * unit))
+
+    bind(units, fn u ->
+      bind(bitstring(length: size), fn bit ->
+        bind(bitstring(length: u), fn bits ->
+          constant(<<bit::bitstring, bits::bitstring>>)
+        end)
+      end)
+    end)
+  end
```
Comment (whatyouhide) [inline lib/stream_data_types.ex]: This looks good, I'd maybe refactor it just slightly:

```
defp generate({:type, _, :binary, [{:integer, _, size}, {:integer, _, unit}]}) do
  rest_length_in_bits = map(non_neg_integer(), &(&1 * unit))

  bind(bitstring(length: size), fn prefix ->
    bind(rest_length_in_bits, fn rest_length ->
      map(bitstring(length: rest_length), &(<<prefix::bitstring, &1::bitstring>>))
    end)
  end)
end
```

Wdyt?

File: lib/stream_data_types.ex [hunk 4]
```diff
@@ -291,7 +312,7 @@ defmodule StreamDataTypes do
   end
 
   defp non_negative_integer() do
-    map(positive_integer(), &(-1 * &1))
+    map(integer(), &abs(&1))
```
Comment (whatyouhide) [inline lib/stream_data_types.ex]: Nitpick: `&abs/1` instead of `&abs(&1)` :)

File: lib/stream_data_types.ex (hunk 3 above)
Comment (NJichev) [reply lib/stream_data_types.ex]: Better names + less nested binds :+1:

**Pull request merged**